### ISSUE TITLE
Bi-directional browsing - to explore

### ISSUE THREAD

**Issue by Klortho** [Klortho/dtd-diagram#4]
Another suggestion from @wendellpiez:

It would be really nice to be able to browse not only down the hierarchy, but up, to show the list of possible parent elements of a node. This would entail a separate flextree that expands to the left.

The way it might work is illustrated here. Start with M, then expand backwards to expose A, B, C. What if the user then wants to see B's kids? We'd have to, I think, sever any connections from B's siblings to M.

I'm not sure what the definitive rule would be for this, though, once the tree starts getting very complicated.

![bidirectional-dtd-diagram](https://cloud.githubusercontent.com/assets/77226/14680273/b802a3aa-06ea-11e6-9450-78257c6948dd.png)

**Comment by wendellpiez** [comment]
The graph represented once we include the 'possible parents' is a very different creature from the simple top-down graph of content models so I think it's fair to say the rules for what should happen as the user browses upwards, remain to be written.

I don't think users should be encouraged to think that A, B and C are siblings here, for example -- since they're not. To shift focus from M up to any of A, B and C is different from the shift from P to to M (wherein Q can be left in place), precisely since P and Q are known to be siblings, whereas A, B and C are not (or rather, are not known to be). While it's true that in graph-space, each of A B and C have "knowable" content models of their own, nothing is actually gained by showing too much info.

This argues that (a) their bubble styling should make it obvious that they are a different sort of node in the tree from any that have been found in a top-down traversal. And (b) that if you want to expand one of them in the 'normal' way (say, B, so as to see F and G) you lose info about the others (A and C). (The user presumably can go back to M and thereby get A and C back.) Maybe it's okay to keep P and Q since they are known to be siblings (are declared together as children of M).

It's true there are some tricky questions here.

**Comment by Klortho** [comment]
From a usability standpoint, I don't think that a lot of flexibility is warranted here (although it's a fun puzzle). I'd think something draconian like, "at most one node can have multiple parents", would be fine. Users would most likely just be interested in one at a time, and that kind of rule wouldn't prevent them from navigating to where they wanted to go.

**Comment by wendellpiez** [comment]
Agreed. Something simple that works is better than something fancy that doesn't.
